core/dbt/events/types.py

```diff
@@ -1927,9 +1927,17 @@ def code(self) -> str:
         return "Z030"
 
     def message(self) -> str:
+
+        class PartialSuccess:
```

Comment:
I agree. I feel like creating a `PartialSuccess` class here feels out of place 🤔 Out of the alternatives you've listed, I think the first would be preferable

Reply:
I started down this path, but two things are giving me pause:
- I would need to make this a nested `import`, to avoid a circular import (since `dbt.artifacts.schemas.results` imports the event system)
- `NodeStatus.Warn` has its string type set to `"warn", and here we have "warning(s)"

```python
class NodeStatus(StrEnum):
    Success = "success"
    Error = "error"
    Fail = "fail"
    Warn = "warn"
    Skipped = "skipped"
    PartialSuccess = "partial success"
    Pass = "pass"
    RuntimeErr = "runtime error"

    def pluralize(self) -> str:
        if self is self.Success or self is self.PartialSuccess or self is self.Pass:
            return f"{self}es"
        elif self is self.Warn:
            return f"warnings" # feels inconsistent
        elif self is self.Skipped:
            return "skipped"
        return f"{self}s"
```

```python
        from dbt.artifacts.schemas.results import NodeStatus

        error_plural = pluralize(self.num_errors, NodeStatus.Error)
        warn_plural = pluralize(self.num_warnings, NodeStatus.Warn)
        partial_success_plural = pluralize(self.num_partial_success, NodeStatus.PartialSuccess)
```

Reply:
> I would need to make this a nested import, to avoid a circular import (since dbt.artifacts.schemas.results imports the event system)
🫠 Right, we haven't finished the decoupling. Woof okay. I'll look in a bit and see if we can break the circular import on the artifacts side.

Reply:
@dbeatty10 It does! That's the `pluralize` method in the code right now

That method says:
- If there's exactly 1 of me, return me (unchanged)
- If there's != 1 of me, and I'm being passed a `string` argument that is actually a class which implements a method named `pluralize`, then use that method to pluralize me; otherwise, just add `s`

So the question here is:
- Should we implement a one-off class for `PartialSuccess` within `EndOfRunSummary` that implements `pluralize`?
- Do I use the actual `NodeStatus` class here, and implement `pluralize`? (problems described above)
- Do I just ignore `dbt_common.events.format.pluralize`, and reimplement that if `count !=1` logic within `EndOfRunSummary`, for "partial success" only?
```python
        error_plural = pluralize(self.num_errors, NodeStatus.Error)
        warn_plural = pluralize(self.num_warnings, NodeStatus.Warn)
        partial_success_plural = f`"""{self.num_partial_success} {"partial success" if self.num_partial_success == 1 else "partial successes"}"""
```

Reply:
```
       partial_success_plural = f"""{self.num_partial_success} {"partial success" if self.num_partial_success == 1 else "partial successes"}"""
```

is ugly, but seems like the fastest and easiest way to solve this in a targeted manner.

If we choose that option, it would solve the problem now, then we could create a tech debt issue to come back and convert it back to `pluralize` once the decoupling part is done.

Obviously totally up to @QMalcolm -- just throwing out some ideass.